Pause and reset channels when load driver

kelvin-cao opened the pull request:
It's an update to the channel initialization sequence.

A reset to the channels is necessary during driver initialization. However, it
could be dangerous to reset a channel with outstanding SE processing by the DMAC
in the background. To avoid this, we pause the channel and wait 60ms to ensure
that all the SE processing has been completed before we reset it during driver
initialization.

ggec62160 commented on switchtec_dma.c:
```diff
@@ -714,25 +738,27 @@ static void switchtec_dma_abort_desc(struct switchtec_dma_chan *swdma_chan)
 		rcu_read_unlock();
 		return;
 	}
 
```
will the channel need wait similar 60ms to let on-going DMA traffic to complete or timeout?

kelvin-cao replied:
As when we do channel stop (halt), the channel is in a known state and is expected to turn to HALTED state within short period of time. So we will depends on halt_channel() which waits for HALTED state for the synchronization.

ggec62160 reviewed (approved):
The implementation of pause_reset_channel() looks good to me.

kelvin-cao committed:
Remove funciton __switchtec_dma_chan_stop

kelvin-cao committed:
Don't do channel reset in function halt_channel()

kelvin-cao committed:
Pause and reset channels when load driver

It's an update to the channel initialization sequence.

A reset to the channels is necessary during driver initialization. However, it
could be dangerous to reset a channel with outstanding SE processing by the DMAC
in the background. To avoid this, we pause the channel and wait 60ms to ensure
that all the SE processing has been completed before we reset it during driver
initialization.
